DumDum/DumDum/Services/TroopService.cs

```diff
@@ -24,47 +25,39 @@ public TroopService(AuthenticateService authService, DumDumService dumService, I
 
         internal GetTroopsResponse ListTroops(string authorization, int kingdomId, out int statusCode)
         {
-            var response = new GetTroopsResponse();
+            
             var player = AuthenticateService.GetUserInfo(new AuthRequest() { Token = authorization });
 
             if (player != null && player.KingdomId == kingdomId)
             {
-                var plyersKingdom = DumDumService.GetKingdomById(player.KingdomId);
-                response.Kingdom = new KingdomResponse
-                {
-                    KingdomId = player.KingdomId,
-                    KingdomName = player.KingdomName,
-                    Ruler = player.Ruler,
-                    Location = new Location() { CoordinateX = plyersKingdom.CoordinateX, CoordinateY = plyersKingdom.CoordinateY }
-                };
-                response.Troops = GetTroops(player.KingdomId);
+                var kingdom = DumDumService.GetKingdomById(player.KingdomId);
+                var response = new GetTroopsResponse(new KingdomResponse(kingdom), GetTroops(player.KingdomId) );
 
                 statusCode = 200;
                 return response;
             }
-
             statusCode = 401;
-            return response;
+            return null;
         }
 
         internal List<TroopsResponse> GetTroops(int kingdomId)
         {
-            return UnitOfWork.Troops.GetTroops(kingdomId);
+            return UnitOfWork.Troops.GetTroops(kingdomId).Result;
         }
 
         internal string UpgradeTroops(string authorization, TroopUpgradeRequest troopUpdateReq, int kingdomId, out int statusCode)
         {
             var player = AuthenticateService.GetUserInfo(new AuthRequest() { Token = authorization });
-            var goldAmount = DumDumService.GetGoldAmountOfKingdom(kingdomId);
             var possibleTroopTypes = UnitOfWork.TroopTypes.PossibleTroopTypesToUpgrade();
 
-            if (troopUpdateReq == null || string.IsNullOrEmpty(troopUpdateReq.Type) || !possibleTroopTypes.Contains(troopUpdateReq.Type.ToLower()))
+            if (troopUpdateReq == null || string.IsNullOrEmpty(troopUpdateReq.Type) || !possibleTroopTypes.Result.Contains(troopUpdateReq.Type.ToLower()))
             {
```

Comment:
v tomhle a podobnym mistech by bylo lepsi kdybys mel to result natom radku predtim. aby si t tady uz nemusel psat